### PR TITLE
feat: support react v18

### PR DESCRIPTION
**Pull request by sharvit**
Add support for react 18.

1. Upgrade react to v18.
2. Add react v18 as a peer dependency.

### CODE REVIEW

File: src/useStep/useStep.ts [hunk 1]
```diff
@@ -9,6 +9,8 @@ interface Helpers {
   setStep: Dispatch<SetStateAction<number>>
 }
 
+type setStepCallbackType = (step: number | ((arg0: number) => number)) => void
```
Comment (sharvit) [inline src/useStep/useStep.ts:12]: The only error I got after the upgrade was related to `step` having `any` type.

**Comment by nmaties** [comment]
@juliencrn Can you please review this and merge it? I am currently getting the following error while running `npm I`:
image

**Comment by secretshardul** [comment]
+1

File: package.json [hunk 2]
```diff
@@ -63,7 +63,7 @@
     "version-bump-prompt": "^6.1.0"
   },
   "peerDependencies": {
-    "react": "^16.9.0 || ^17.0.0"
+    "react": "^16.9.0 || ^17.0.0 || ^18.0.0"
```
Comment (juliencrn) [inline package.json:66]: 👍

File: package.json [hunk 3]
```diff
@@ -43,8 +43,8 @@
     "@testing-library/react-hooks": "^7.0.2",
     "@types/jest": "^29.0.3",
     "@types/node": "^18.7.18",
-    "@types/react": "^17.0.38",
-    "@types/react-dom": ">=16.9.0",
+    "@types/react": "^18.0.0",
+    "@types/react-dom": "^18.0.0",
```
Comment (juliencrn) [inline package.json:47]: This works when used by react@18 projects, but it doesn't work else on my tests on a Gatsby powered react@17 website.

**Review by juliencrn — COMMENTED** [review]
Hi @sharvit, 

Thank you for this wanted feature. Unfortunately, it doesn't work as expected and meanwhile, I've worked on it, too. So I will close this PR, but I've used part of your code and added you (and a ref to this PR) in the commit message, the changelog and the release note. 

Thanks 😊

**Comment by juliencrn** [comment]
Added in usehooks-ts@2.7.1 (https://github.com/juliencrn/usehooks-ts/releases/tag/usehooks-ts%402.7.1)

Thanks again! :)

**Comment by sharvit** [comment]
Thanks for the quick replay and fix @juliencrn, the new release works perfectly for me :+1:

**Pull request closed without merging**